hbase-client/src/main/java/org/apache/hadoop/hbase/exceptions/ClientExceptionsUtil.java

```diff
@@ -59,18 +58,28 @@ public static boolean isMetaClearingException(Throwable cur) {
     if (cur == null) {
       return true;
     }
-    return !isSpecialException(cur) || (cur instanceof RegionMovedException)
-        || cur instanceof NotServingRegionException;
+    return !isMetaCachePreservingException(cur);
   }
 
-  public static boolean isSpecialException(Throwable cur) {
-    return (cur instanceof RegionMovedException || cur instanceof RegionOpeningException
-        || cur instanceof RegionTooBusyException || cur instanceof RpcThrottlingException
-        || cur instanceof MultiActionResultTooLarge || cur instanceof RetryImmediatelyException
-        || cur instanceof CallQueueTooBigException || cur instanceof CallDroppedException
-        || cur instanceof NotServingRegionException || cur instanceof RequestTooBigException);
+  public static boolean isRegionServerOverloadedException(Throwable t) {
+    t = findException(t);
+    return isInstanceOfRegionServerOverloadedException(t);
+  }
+
+  private static boolean isInstanceOfRegionServerOverloadedException(Throwable t) {
+    return t instanceof CallQueueTooBigException || t instanceof CallDroppedException;
   }
 
+  private static boolean isMetaCachePreservingException(Throwable t) {
+    return t instanceof RegionOpeningException || t instanceof RegionTooBusyException
+        || t instanceof RpcThrottlingException || t instanceof RetryImmediatelyException
+        || t instanceof RequestTooBigException;
```

Comment:
So, this list here is smaller than the list above in the old isSpecialException. I looked at one or two of the differences. It seems like superclasses are listed here and you drop the mention of subclasses. Is that the case? Just a little worried we're not catching something here that we used to.

Reply:
Yes, just remove subclasses, you mean the detailed messages in the subclass?